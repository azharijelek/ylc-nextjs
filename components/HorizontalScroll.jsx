import PropTypes from 'prop-types'

export default function HorizontalScroll(props) {
  return (
    <>
      <div className="hs">{props.children}</div>

      <style jsx>{`
        .hs {
          display: flex;
          flex-wrap: nowrap;
          overflow-x: auto;
          -webkit-overflow-scrolling: touch;
          &::-webkit-scrollbar {
            display: none;
          }
          &:before,
          &:after {
            content: '';
          }
        }
      `}</style>
    </>
  )
}

HorizontalScroll.propTypes = {
  children: PropTypes.any
}
