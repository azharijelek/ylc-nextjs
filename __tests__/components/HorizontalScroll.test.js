import React from 'react'
import Adapter from 'enzyme-adapter-react-16'
import { shallow, configure } from 'enzyme'
import HorizontalScroll from '@/components/HorizontalScroll'

configure({ adapter: new Adapter() })

describe('<HorizontalScroll/>', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<HorizontalScroll />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should has .hs class', () => {
    const wrapper = shallow(<HorizontalScroll />)
    expect(wrapper.find('.hs')).toHaveLength(1)
  })
})
