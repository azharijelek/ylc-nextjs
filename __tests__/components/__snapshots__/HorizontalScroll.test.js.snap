// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HorizontalScroll/> renders correctly 1`] = `ShallowWrapper {}`;
